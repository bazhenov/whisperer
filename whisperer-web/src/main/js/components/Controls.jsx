import React from 'react'

const Controls = (props) => {
	const { isListening, startListening, stopListening, clearMessages, hasMessages } = props;

	return <div id="control-buttons">
		{isListening ?
			<a href="#" onClick={(e) => { stopListening(); e.preventDefault() }} className="btn btn-default">STOP</a> :
			<a href="#" onClick={(e) => { startListening(); e.preventDefault() }} className="btn btn-default">LISTEN</a>}
		{hasMessages ?
			<a href="#" onClick={(e) => { clearMessages(); e.preventDefault() }} className="btn btn-danger">CLEAR</a> :
			''
		}
	</div>
};

export default Controls;