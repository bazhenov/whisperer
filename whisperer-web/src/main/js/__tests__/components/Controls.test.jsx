import React from "react";
import Controls from "../../components/Controls.jsx";
import renderer from "react-test-renderer";
import {mount} from "enzyme";

describe('Controls', () => {
	it('rendering', () => {
		[
			{ isListening: false, hasMessages: false },
			{ isListening: true, hasMessages: false },
			{ isListening: false, hasMessages: true },
			{ isListening: true, hasMessages: true }
		].forEach(props => {
			const tree = renderer.create(
				<Controls { ...props } startListening={() => {}} stopListening={() => {}} clearMessages={() => {}} />
			).toJSON();
			expect(tree).toMatchSnapshot();
		});
	});

	describe('click handlers', () => {
		const props = {
			isListening: false,
			hasMessages: false,
			startListening: () => {},
			stopListening: () => {},
			clearMessages: () => {}
		};

		const simulateClick = (buttonText, props) => {
			mount(
				<Controls { ...props } />
			).find('a').filterWhere(link => link.text() === buttonText).first().simulate('click');
		};

		it('startListening', done => {
			simulateClick('LISTEN', { ...props, startListening: () => done() });
		});

		it('stopListening', done => {
			simulateClick('STOP', { ...props, isListening: true, stopListening: () => done() });
		});

		it('stopListening', done => {
			simulateClick('CLEAR', { ...props, hasMessages: true, clearMessages: () => done() });
		});
	});
});