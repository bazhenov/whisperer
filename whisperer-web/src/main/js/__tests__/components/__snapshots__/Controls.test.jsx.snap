// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls rendering 1`] = `
<div
  id="control-buttons"
>
  <a
    className="btn btn-default"
    href="#"
    onClick={[Function]}
  >
    LISTEN
  </a>
  
</div>
`;

exports[`Controls rendering 2`] = `
<div
  id="control-buttons"
>
  <a
    className="btn btn-default"
    href="#"
    onClick={[Function]}
  >
    STOP
  </a>
  
</div>
`;

exports[`Controls rendering 3`] = `
<div
  id="control-buttons"
>
  <a
    className="btn btn-default"
    href="#"
    onClick={[Function]}
  >
    LISTEN
  </a>
  <a
    className="btn btn-danger"
    href="#"
    onClick={[Function]}
  >
    CLEAR
  </a>
</div>
`;

exports[`Controls rendering 4`] = `
<div
  id="control-buttons"
>
  <a
    className="btn btn-default"
    href="#"
    onClick={[Function]}
  >
    STOP
  </a>
  <a
    className="btn btn-danger"
    href="#"
    onClick={[Function]}
  >
    CLEAR
  </a>
</div>
`;
